lib/logthrsource/logthrfetcherdrv.c

```diff
@@ -26,10 +26,10 @@
 #include "messages.h"
 
 void
-log_threaded_fetcher_driver_set_fetch_no_data_delay(LogDriver *s, time_t no_data_delay)
+log_threaded_fetcher_driver_set_fetch_no_data_delay(LogDriver *s, gdouble no_data_delay)
 {
   LogThreadedFetcherDriver *self = (LogThreadedFetcherDriver *) s;
-  self->no_data_delay = no_data_delay;
+  self->no_data_delay = (time_t) (no_data_delay * 1000 * 1000 * 1000);
```

Comment:
`time_t` is usually for storing seconds since 1970. I am a little afraid that as it's size is implementation dependent, we might not fit into it. I think we should change `no_data_delay` to `gint64`.

Reply:
Yes, you are absolutely right.